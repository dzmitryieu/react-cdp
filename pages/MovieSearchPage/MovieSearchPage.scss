@import "../../src/utils/constants.scss";

.wrapper {
  background-image: url(../../src/assets/images/cover-image.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
}

.section {
  padding: 0 60px 50px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 600px;
  position: relative;
  background: rgba(0, 0, 0, 0.65);
}

.img {
  position: absolute;
  top: 0;
  left: 0;
  width: 100vw;
}

@media screen and (max-width: $tablet-size-screen) {
  .section {
    padding: 0 20px;
  }
}
