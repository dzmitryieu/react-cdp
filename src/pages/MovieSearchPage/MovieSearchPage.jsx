import React from 'react';
import PropTypes from 'prop-types';
import styles from './MovieSearchPage.scss';
import Header from '../../components/Header';
import Search from '../../components/Search';
import '../../assets/images/cover-image.jpg';

const MovieSearchPage = ({ isShowSearchPage }) => {
  if (!isShowSearchPage) return null;
  return (
    <div id="MovieSearchPage" className={styles.wrapper}>
      <section className={styles.section}>
        <Header />
        <Search />
      </section>
    </div>
  );
};

MovieSearchPage.propTypes = {
  isShowSearchPage: PropTypes.bool.isRequired,
};

export default MovieSearchPage;
