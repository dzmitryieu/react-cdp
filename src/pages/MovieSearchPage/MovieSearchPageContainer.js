import { connect } from 'react-redux';
import MovieSearchPage from './MovieSearchPage';
import { isShowSearchPage } from '../../modules/film/filmSelectors';


const mapStateToProps = state => ({
  isShowSearchPage: isShowSearchPage(state),
});

export default connect(mapStateToProps)(MovieSearchPage);
