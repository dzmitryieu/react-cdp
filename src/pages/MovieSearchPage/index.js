import MovieSearchPage from './MovieSearchPageContainer';

export default MovieSearchPage;
