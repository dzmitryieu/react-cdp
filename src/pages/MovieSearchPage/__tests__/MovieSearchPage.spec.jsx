import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Testrenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import MovieSearchPage from '../MovieSearchPage';
import MovieSearchPageContainer from '../index';

Enzyme.configure({ adapter: new Adapter() });
const renderer = new ShallowRenderer();

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(
  {
    persistedStore: {
      showSearchPage: true,
    },
    notPersistedStore: {
      searchedWord: '',
    },
  },
);

const fetchAllFilms = jest.fn();
const makeMainFilm = jest.fn();
const mockFunc = { fetchAllFilms, makeMainFilm };


describe('MovieSearchPage Snapshot', () => {
  test('renders', () => {
    const component = renderer.render(
      <MovieSearchPage isShowSearchPage={false} />,
    );
    expect(component).toMatchSnapshot();
  });
  test('no render if it is not needed', () => {
    const component = renderer.render(
      <MovieSearchPage isShowSearchPage />,
    );
    expect(component).toMatchSnapshot();
  });
  test('with store render', () => {
    const component = Testrenderer.create(
      <Provider store={store}>
        <MovieSearchPageContainer {...mockFunc} />
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
