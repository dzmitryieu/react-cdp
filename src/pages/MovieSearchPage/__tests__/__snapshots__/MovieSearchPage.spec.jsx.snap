// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSearchPage Snapshot no render if it is not needed 1`] = `
<div
  className="wrapper"
  id="MovieSearchPage"
>
  <section
    className="section"
  >
    <Connect(Header) />
    <withRouter(Connect(Search)) />
  </section>
</div>
`;

exports[`MovieSearchPage Snapshot render with store 1`] = `
<div
  className="wrapper"
  id="MovieSearchPage"
>
  <section
    className="section"
  >
    <header
      className="header"
    >
      <a
        className="link"
        href="/"
        onClick={[Function]}
      >
        <p
          className="logo"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="presentation"
        >
          netflixroulette
        </p>
      </a>
    </header>
    <div
      className="searchWrapper"
    >
      <p
        className="searchText"
      >
        find your movie
      </p>
      <input
        className="search"
        onChange={[Function]}
        placeholder="search film"
        type="text"
        value=""
      />
      <div
        className="buttonsWrapper"
      >
        <div
          className="btnWrapper"
        >
          <h3
            className="searchTitle"
          >
            SEARCH BY
          </h3>
          <button
            className="button  red null"
            data-attr="title"
            id={null}
            onClick={[Function]}
            type="button"
          >
            title
          </button>
          <button
            className="button  black null"
            data-attr="genres"
            id={null}
            onClick={[Function]}
            type="button"
          >
            genre
          </button>
        </div>
        <button
          className="button big red null"
          data-attr={null}
          id={null}
          onClick={[Function]}
          type="button"
        >
          SEARCH
        </button>
      </div>
    </div>
  </section>
</div>
`;

exports[`MovieSearchPage Snapshot renders 1`] = `null`;
